checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsTransfer.java

```diff
@@ -61,6 +62,47 @@ public CalledMethodsTransfer(CalledMethodsAnalysis analysis) {
         ((CalledMethodsAnnotatedTypeFactory) atypeFactory).calledMethodsValueElement;
   }
 
+  /**
+   * The Called Methods Checker is one of the sub-checkers of the Resource Leak Checker. When we
+   * enable -Ainfer for the Resource Leak Checker, whole-program inference (wpi) is performed for
+   * all the sub-checkers. However, our mechanism for inferring annotations for the resource leak
+   * checker is different. It relies on pattern matching instead. The wpi results for this inference
+   * are not useful and only end up slowing down the convergence of the algorithm for Resource Leak
+   * Inference.
+   *
+   * @param tree a tree
+   * @return false if Resource Leak Checker is running as one of the upstream checkers, otherwise
+   *     returns the result of the super call
+   */
+  @Override
+  protected boolean shouldPerformWholeProgramInference(Tree tree) {
+    if (isResourceLeakCheckerEnabled()) {
+      return false;
+    }
+    return super.shouldPerformWholeProgramInference(tree);
+  }
+
+  /**
+   * The Called Methods Checker is one of the sub-checkers of the Resource Leak Checker. When we
+   * enable -Ainfer for the Resource Leak Checker, whole-program inference (wpi) is performed for
+   * all the sub-checkers. However, our mechanism for inferring annotations for the resource leak
+   * checker is different. It relies on pattern matching instead. The wpi results for this inference
+   * are not useful and only end up slowing down the convergence of the algorithm for Resource Leak
+   * Inference.
+   *
+   * @param expressionTree a tree
+   * @param lhsTree its element
+   * @return false if Resource Leak Checker is running as one of the upstream checkers, otherwise
+   *     returns the result of the super call
+   */
+  @Override
+  protected boolean shouldPerformWholeProgramInference(Tree expressionTree, Tree lhsTree) {
+    if (isResourceLeakCheckerEnabled()) {
```

Comment:
I would expect users to be able to configure whether they use WPI or your inference logic via a command-line option that is independent of whether the RLC is running or not. That is, I'd expect it to be possible to:
1. run the RLC with your inference system
2. run the RLC with WPI
3. run the Called Methods Checker/Must Call Checker with your inference system
4. run the Called Methods Checker/Must Call Checker with WPI

I think this approach only permits users to do items 1 and 4 from this list. At the very least, option 2 needs to be possible (I think I will want it in the future to test some WPI improvements that would make WPI + the RLC work better together). I suggest adding a command-line option to the RLC that enables your inference system (maybe it defaults to "on" when -Ainfer is passed, but can be toggled "off").